packages/amplify-graphql-relational-transformer/src/resolvers.ts

```diff
@@ -0,0 +1,285 @@
+import {
+  applyCompositeKeyConditionExpression,
+  applyKeyConditionExpression,
+  attributeTypeFromScalar,
+  ModelResourceIDs,
+  NONE_VALUE,
+  ResourceConstants,
+  toCamelCase,
+} from 'graphql-transformer-common';
+import { HasManyDirectiveConfiguration, HasOneDirectiveConfiguration } from './types';
+import { getConnectionAttributeName } from './utils';
+
+export function makeGetItemConnectionWithKeyResolver(config: HasOneDirectiveConfiguration, ctx: TransformerContextProvider) {
+  const { connectionFields, field, fields, object, relatedType, relatedTypeIndex } = config;
+  assert(relatedTypeIndex.length > 0);
+  const localFields = fields.length > 0 ? fields : connectionFields;
+  const table = getTable(ctx, relatedType);
+  const { keySchema } = table as any;
+  const dataSource = ctx.api.host.getDataSource(`${relatedType.name.value}Table`);
+  const partitionKeyName = keySchema[0].attributeName;
+  const keyObj = {
+    [partitionKeyName]: ref(`util.dynamodb.toDynamoDBJson($util.defaultIfNullOrBlank($ctx.source.${localFields[0]}, "${NONE_VALUE}"))`),
+  };
+
+  // Add a composite sort key or simple sort key if there is one.
+  if (relatedTypeIndex.length > 2) {
+    const rangeKeyFields = localFields.slice(1);
+    const sortKeyName = keySchema[1].attributeName;
+    const condensedSortKeyValue = condenseRangeKey(rangeKeyFields.map(keyField => `\${ctx.source.${keyField}}`));
+
+    keyObj[sortKeyName] = ref(`util.dynamodb.toDynamoDBJson($util.defaultIfNullOrBlank("${condensedSortKeyValue}", "${NONE_VALUE}"))`);
+  } else if (relatedTypeIndex.length === 2) {
+    const sortKeyName = keySchema[1].attributeName;
+
+    keyObj[sortKeyName] = ref(`util.dynamodb.toDynamoDBJson($util.defaultIfNullOrBlank($ctx.source.${localFields[1]}, "${NONE_VALUE}"))`);
+  }
+
+  const resolver = ctx.resolvers.generateQueryResolver(
+    object.name.value,
+    field.name.value,
+    dataSource as any,
+    MappingTemplate.s3MappingTemplateFromString(
+      print(
+        ifElse(
+          or(localFields.map(f => raw(`$util.isNull($ctx.source.${f})`))),
+          raw('#return'),
+          compoundExpression([
+            DynamoDBMappingTemplate.getItem({
+              key: obj(keyObj),
+            }),
+          ]),
+        ),
+      ),
+      `${object.name.value}.${field.name.value}.req.vtl`,
+    ),
+    MappingTemplate.s3MappingTemplateFromString(
+      print(DynamoDBMappingTemplate.dynamoDBResponse(false)),
+      `${object.name.value}.${field.name.value}.res.vtl`,
+    ),
+  );
+
+  resolver.mapToStack(table.stack);
+  ctx.resolvers.addResolver(object.name.value, field.name.value, resolver);
+}
+
+export function makeQueryConnectionWithKeyResolver(config: HasManyDirectiveConfiguration, ctx: TransformerContextProvider) {
+  const { connectionFields, field, fields, indexName, limit, object, relatedType } = config;
+  const table = getTable(ctx, relatedType);
+  const dataSource = ctx.api.host.getDataSource(`${relatedType.name.value}Table`);
```

Comment:
We should have the table name format centralized somewhere, I think in `@model` it uses DS? Maybe also have this in mapping template or core.